javascript/index.js

```diff
@@ -24,17 +24,89 @@
     document.querySelector("#mashedPotatoes").innerHTML += `<li>${step5}</li>`;
     document.querySelector("#mashedPotatoesImg").removeAttribute("hidden");
   }, (error) => console.log(error));
+  */
 
 
 
 // Iteration 1 - using callbacks
-// ...
+  getInstruction('mashedPotatoes', 0, (step0) => {
+    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step0}</li>`
+    getInstruction('mashedPotatoes', 1, (step0) => {
+      document.querySelector("#mashedPotatoes").innerHTML += `<li>${step0}</li>`})
+      getInstruction('mashedPotatoes', 2, (step0) => {
+        document.querySelector("#mashedPotatoes").innerHTML += `<li>${step0}</li>`})
+        getInstruction('mashedPotatoes', 3, (step0) => {
+          document.querySelector("#mashedPotatoes").innerHTML += `<li>${step0}</li>`})
+          getInstruction('mashedPotatoes', 4, (step0) => {
+            document.querySelector("#mashedPotatoes").innerHTML += `<li>${step0}</li>`})
+          })
+        
 
 // Iteration 2 - using promises
-// ...
+obtainInstruction("steak", 0)
+.then((step0)=>{
+  document.querySelector("#steak").innerHTML += `<li>${step0}</li>`
+  return obtainInstruction("steak", 1)
+}).then((step2)=>{
+  document.querySelector("#steak").innerHTML += `<li>${step2}</li>`
+    return obtainInstruction("steak", 2)
+}).then((step3)=>{
+  document.querySelector("#steak").innerHTML += `<li>${step3}</li>`
+    return obtainInstruction("steak", 3)
+}).then((step4)=>{
+  document.querySelector("#steak").innerHTML += `<li>${step4}</li>`
+    return obtainInstruction("steak", 4)
+}).then((step5)=>{
+  document.querySelector("#steak").innerHTML += `<li>${step5}</li>`
+    return obtainInstruction("steak", 5)
+}).then((step6)=>{
+  document.querySelector("#steak").innerHTML += `<li>${step6}</li>`
+    return obtainInstruction("steak", 6)
+}).then((step7)=>{
+  document.querySelector("#steak").innerHTML += `<li>${step7}</li>`
+    return obtainInstruction("steak", 7)
+}).then((step8)=>{
+  document.querySelector("#steak").innerHTML += `<li>${step8}</li>`
+}).catch((error)=>{
+console.log("error", error)
+ 
+}).finally(()=>{
```

Comment:
Súper que aplicaste el finally, sólo recuerda que el código dentro de si este se ejecuta independientemente si la promesa se ejecuta o no. Es decir: `Stake is ready!` va a aparecer aunque las promesas anteriores no se cumplan.